docs/source/install.rst

```diff
@@ -307,6 +313,7 @@ If you have installed CUDA on the non-default directory or multiple CUDA version
 
 CuPy uses the first CUDA installation directory found by the following order.
 
+#. ``cuda-pathfinder``'s `documented search order <https://nvidia.github.io/cuda-python/cuda-pathfinder/latest/generated/cuda.pathfinder.load_nvidia_dynamic_lib.html>`_.
```

Comment:
This can be out of the scope of this PR, but this description is slightly inaccurate. We rely on several aspects of CTK and their discovery strategy is a bit cluttered:

* CTK shared libraries
  * Fully rely on ``cuda-pathfinder``'s documented search order, after this PR get merged
* CTK headers
  1. `$DISCOVERED_CUDA_PATH/include`
  2. NVIDIA Pip packages: `/path/to/site-packages/nvidia/cuda_runtime/include` (for CUDA 12) or `/path/to/site-packages/nvidia/cuda/cu13/include` (for CUDA 13)
* NVCC
  1. `$NVCC` env var
  2. `$DISCOVERED_CUDA_PATH/bin/nccl`
  3. OS default strategy (`$PATH`)

For CTK headers I think it's better to migrate to `cuda.pathfinder.find_nvidia_header_directory` at some point.

Reply:
Yes I'll work on the header bits after this is merged.